plugins/rbac/src/components/CreateRole/PermissionPoliciesForm.tsx

```diff
@@ -121,33 +126,50 @@ export const PermissionPoliciesForm = ({
         multiple permission policies using +Add option.
       </FormHelperText>
       <br />
-      <div className={classes.permissionPoliciesForm}>
-        {permissionPoliciesRows.map((pp, index) => (
-          <PermissionPoliciesFormRow
-            key={index}
-            permissionPoliciesRowError={
-              permissionPoliciesRowsError?.[index] ?? {}
-            }
-            rowName={`permissionPoliciesRows[${index}]`}
-            permissionPoliciesRowData={pp}
-            permissionPoliciesData={permissionPoliciesData}
-            rowCount={permissionPoliciesRows.length}
-            onChangePlugin={(plugin: string) => onChangePlugin(plugin, index)}
-            onChangePermission={(permission: string, policies?: string[]) =>
-              onChangePermission(permission, index, policies)
-            }
-            onChangePolicy={(isChecked: boolean, policyIndex: number) =>
-              onChangePolicy(isChecked, policyIndex, index)
-            }
-            onRemove={() => onRowRemove(index)}
-            handleBlur={handleBlur}
-          />
-        ))}
-        <Button className={classes.addButton} size="small" onClick={onRowAdd}>
-          <AddIcon />
-          Add
-        </Button>
-      </div>
+      {permissionPoliciesLoading ? (
+        <Progress />
```

Comment:
@divyanshiGupta  I am checking here already if the permission policies are still loading

Reply:
Yeah but it's not being shown 🤔 can you check why is that? @debsmita1

Reply:
Maybe because the value of `loading` updates from true to false way faster

Reply:
Yeah. So to handle this we will have to show the progress bar for a minimum of 1-2 secs so that the progress bar is shown correctly and if loading takes more than that we show the progress bar for the entire duration. LMK if you have some other idea.

Reply:
I think we should definitely have the loading/progress bar handled but at the same time, we should show the progress bar only if it's actually loading be it for a second or more, as loading gets false we should render the data.

@debsmita1 pls test with throttling in the browser.

Reply:
I changed the network preset to 'Fast 3G' and the ProgressBar is visible
This means that the progress bar does show up when there's a delay in fetching the results

https://github.com/janus-idp/backstage-plugins/assets/22490998/fbd0702c-2d4b-4caf-a3c8-4a8ba09c8fbd

Reply:
yeah, the issue is because the loading is taking probably like some ms but I pointed this out because it shows nothing for those ms and then the table is rendered. Definitely if the loading takes more time the progress bar will be shown and that is where I see the inconsistency. But I suppose its not a big issue right now and we can always come back to it later if its pointed out.